File: src/main/java/ru/hollowhorizon/hollowstory/common/entities/ModEntities.java
package ru.hollowhorizon.hollowstory.common.entities;

import net.minecraft.entity.EntityClassification;
import net.minecraft.entity.EntityType;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import ru.hollowhorizon.hc.api.registy.HollowRegister;
import ru.hollowhorizon.hollowstory.client.render.enitites.SindyRenderer;

public class ModEntities {
    public static final DeferredRegister<EntityType<?>> ENTITIES = DeferredRegister.create(ForgeRegistries.ENTITIES, "hollowstory");
    public static final RegistryObject<EntityType<NPCEntity>> NPC_ENTITY = ENTITIES.register("npc_entity", () -> EntityType.Builder.of(
            (EntityType.IFactory<NPCEntity>) NPCEntity::new,
            EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowstory:npc_entity"));
    public static final RegistryObject<EntityType<ThomasNPC>> THOMAS = ENTITIES.register("thomas", () -> EntityType.Builder.of(
            ThomasNPC::new,
            EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowstory:thomas"));

    @HollowRegister(renderer = SindyRenderer.class)
    public static final EntityType<SindyNPC> SINDY = EntityType.Builder.of(SindyNPC::new, EntityClassification.CREATURE).sized(0.6F, 1.8F).build("hollowstory:sindy");

    public static void register() {

    }
}

File: runtime/src/main/java/ru/hollowhorizon/hollowengine/common/registry/ModEntities.kt
package ru.hollowhorizon.hollowengine.common.registry

import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.MobCategory
import ru.hollowhorizon.hollowengine.HollowEngine
import ru.hollowhorizon.hollowengine.common.entities.NpcEntity
import ru.hollowhorizon.hollowengine.common.entities.SeatEntity
import ru.hollowhorizon.hollowengine.common.events.SubscribeEvent
import ru.hollowhorizon.hollowengine.common.events.registry.RegisterEntityAttributesEvent

object ModEntities : HollowRegistry(HollowEngine.MODID) {
    val NPC_ENTITY: EntityType<NpcEntity> by register("npc_entity") {
        EntityType.Builder.of(::NpcEntity, MobCategory.CREATURE).sized(0.6f, 1.8f).build("npc_entity")
    }

    val SEAT: EntityType<SeatEntity> by register("seat") {
        EntityType.Builder.of(::SeatEntity, MobCategory.CREATURE).sized(0.0f, 0.0f).build("seat")
    }
}

@SubscribeEvent
fun onRegisterAttributes(event: RegisterEntityAttributesEvent) {
    event.register(ModEntities.NPC_ENTITY, NpcEntity.createAttributes().build())
    event.register(ModEntities.SEAT, LivingEntity.createLivingAttributes().build())
}